allennlp/modules/dot_product_matrix_attention.py

```diff
@@ -0,0 +1,26 @@
+"""
+A ``Module`` that takes two matrices as input and returns a matrix of attentions.
+"""
+import math
+
+import torch
+
+from allennlp.common import Params
+from allennlp.modules.matrix_attention import MatrixAttention
+from overrides import overrides
+
+@MatrixAttention.register("dot_product_matrix_attention")
```

Comment:
These names don't need to have `_matrix_attention` in them, because that'll be obvious based on context.  E.g., here it's inside of `MatrixAttention.register()`, so it's obviously matrix attention, and where you give the parameter in the configuration file it's inside of an "attention" block (or similar).